### PR TITLE
Nil check for prototype

### PR DESCRIPTION
**Pull request by rbindra**
Currently `debug.traceback` fails with error `"An error occurred: genghis flow failed to execute with interface conversion: lua.value is nil, not *lua.luaClosure"`. This PR attempts to fix this. 

To 🎩 how `debug.traceback` will look in action once this is fixed ->

1. Go to `genghis` repo and on the terminal run `go get github.com/Shopify/go-lua@5d21fc2948423a50bfe5fbc78dc49d2025671971` (This SHA is this commit id)
2. Then run `go mod vendor` and then `dev build`
3. Go to `genghis-flows` repo and copy the binary generated (in genghis/bin/genghis) from previous step to this repo  (in .dev/binaries/genghis) by running  `cp ../genghis/bin/genghis .dev/binaries/genghis `
4. In ` genghis-flows` create a file `test_stacktrace.lua` with following code snippet: 
```
local debug = require 'debug'

function a ()
  print(debug.traceback())
  end

function b ()
  a()
end  

function c ()
  b()
end

c ()
```
5. Then create a file `test_stacktrace.json` with the path to `test_stacktrace.lua`
```
{
  "path": "test_stacktrace.lua"
}
```
6. Run `dev exec test_stacktrace.json` and see the stacktrace information.
7. The stacktrace should show something likes this:
```stack traceback:
	./platform-conditioning/test_stacktrace.lua:5: in function <./platform-conditioning/test_stacktrace.lua:3>
	./platform-conditioning/test_stacktrace.lua:13: in function <./platform-conditioning/test_stacktrace.lua:11>
	./platform-conditioning/test_stacktrace.lua:18: in function <./platform-conditioning/test_stacktrace.lua:16>
	./platform-conditioning/test_stacktrace.lua:21: in main chunk
	[Go]: in function 'require'
	[string "require "platform-conditioning.test_stacktrace"..."]:1: in main chunk
```

### CODE REVIEW

File: debug.go [hunk 1]
```diff
@@ -11,10 +11,13 @@ type Frame *callInfo
...
 func (l *State) resetHookCount() { l.hookCount = l.baseHookCount }
 func (l *State) prototype(ci *callInfo) *prototype {
+	if l.stack[ci.function] == nil {
+		return nil
+	}
 	return l.stack[ci.function].(*luaClosure).prototype
 }
 func (l *State) currentLine(ci *callInfo) int {
-	return int(l.prototype(ci).lineInfo[ci.savedPC - 1])
+	return int(l.prototype(ci).lineInfo[ci.savedPC-1])
```
Comment (bjedrocha) [inline debug.go:20]: Is this change actually required?

**Review by bjedrocha — COMMENTED** [review]
Code LGTM although I don't have a lot of context here. Have you been able to determine why `l.stack[ci.function]` is `nil` in the first place?

File: debug.go (hunk 1 above)
Comment (rbindra) [reply debug.go]: Nah, got added by mistake

File: debug.go [hunk 2]
```diff
@@ -228,6 +231,9 @@ func functionInfo(p Debug, f closure) (d Debug) {
 func (l *State) functionName(ci *callInfo) (name, kind string) {
```
Comment (fbogsany) [inline debug.go:231]:
```suggestion
func (l *State) functionName(ci *callInfo) (name, kind string) {
	if ci == &l.baseCallInfo {
		return
	}
```

It's a little complicated, but this case is triggered by
https://github.com/Shopify/go-lua/blob/b252935ae7ee132380e874826ecfadb3b5515afe/debug.go#L364-L365
when we reach the start of the stack. The initial stack frame is "fake" - a sentinel representing the initial call from Go to Lua. We check for the sentinel when walking the stack:
https://github.com/Shopify/go-lua/blob/b252935ae7ee132380e874826ecfadb3b5515afe/debug.go#L198-L202
but we reach past it in `Info`.

**Review by fbogsany — CHANGES_REQUESTED** [review]
You shouldn't need the `nil` checks - checking for the sentinel frame in `functionName` is sufficient.

File: debug.go (hunk 2 above)
Comment (rbindra) [reply debug.go:231]: Awesome! Thanks @fbogsany. I wasn't aware of `baseCallInfo`

**Review by fbogsany — APPROVED** [review]

**Comment by joeytepp** [comment]
Going to merge this 👍
